Libraries/LibUnicode/Segmenter.cpp

```diff
@@ -183,14 +183,14 @@ class SegmenterImpl : public Segmenter {
 
         return m_segmented_text.visit(
             [&](String const& text) -> Optional<i32> {
-                if (boundary >= text.byte_count())
+                if (boundary > text.byte_count())
```

Comment:
Is this not guaranteed UB if `icu_boundary == text.byte_count()`?
The expansion of `U8_SET_CP_START` will contain, essentially:
```c++
text.bytes().data()[text.byte_count()]
```

Reply:
To be honest, the extent of the checking I did was to run all tests with sanitizers on and all looked good. This definitely looks wrong though :thinking: